src/org/usfirst/frc/team1076/robot/SonarReciever.java

```diff
@@ -0,0 +1,37 @@
+package org.usfirst.frc.team1076.robot;
+
+import java.io.IOException;
+import java.net.DatagramPacket;
+import java.net.DatagramSocket;
+import java.net.InetSocketAddress;
+import java.net.SocketException;
+
+import org.json.JSONObject;
+import org.json.JSONTokener;
+
+public class SonarReciever {
+	DatagramSocket socket;
+	int sonarDistance;
+
+	public SonarReciever(String ip, int portNumber) throws SocketException {
+		socket = new DatagramSocket(null);
+		InetSocketAddress address = new InetSocketAddress(ip, portNumber);
+		socket.bind(address);
+	}
+
+	public void recieve() throws IOException {
+
+		byte[] buffer = new byte[512];
+		DatagramPacket packet = new DatagramPacket(buffer, buffer.length);
+		socket.receive(packet);
```

Comment:
This is a blocking receive, if there's no data to receive this shouldn't block and should instead not change anything.